### PR TITLE
Schema for images

### PR DESCRIPTION
**Pull request by jnolis**
Like the Resources PR, this lets you use a file to specify what an image/version is.

### CODE REVIEW

**Comment by jnolis** [comment]
> approved regardless of the answer to this question - what if the image supports python and r?

Maybe the field shouldn't be "python" or "r" but "jupyterlab", "rstudio", "vscode", and you can specify multiple.

Wait that just feels like a legit good idea because then our UI could let you specify the IDE you want (if any) and then which image that works for that UI

**Comment by forana** [comment]
> Wait that just feels like a legit good idea because then our UI could let you specify the IDE you want (if any) and then which image that works for that UI

How about something like:

```json
supportedIDEs: ["jupyterlab", "rstudio"]
```

**Comment by hhuuggoo** [comment]
> > Wait that just feels like a legit good idea because then our UI could let you specify the IDE you want (if any) and then which image that works for that UI
> 
> How about something like:
> 
> ```js
> {
>   /* ... */
>   "supportedIDEs": ["jupyterlab", "rstudio"]
> }
> ```

On that note - we often get errors of people trying to use Dask when dask is not installed. so expanding that list to encompass dask would be a useful thing.

**Comment by forana** [comment]
> > > Wait that just feels like a legit good idea because then our UI could let you specify the IDE you want (if any) and then which image that works for that UI
> > 
> > 
> > How about something like:
> > ```js
> > {
> >   /* ... */
> >   "supportedIDEs": ["jupyterlab", "rstudio"]
> > }
> > ```
> 
> On that note - we often get errors of people trying to use Dask when dask is not installed. so expanding that list to encompass dask would be a useful thing.

"List of things this image supports" is a concept that could help us deprecate a lot of funky code on the backend.

**Comment by jnolis** [comment]
If we turn it into a list of things the image supports I think beyond dask we could also include "ssh" and even "gpu" (although that last one might have weird implications)

File: images/example.json [hunk 1]
```diff
@@ -0,0 +1,12 @@
+{
```
Comment (jsignell) [inline images/example.json:1]: Just a note that the pattern in resources uses nesting pretty agressively. I'd like to do that here as well. So there would be two dicts: "image" and "version".

Comment (jsignell) [reply images/example.json:1]: Actually it would be `versions` and it would be a list of version dicts. That woudl be best.

**Comment by jnolis** [comment]
I just updated this with our latest recipe thinking. The only weird note is I think we should change `version` (the field that denotes the saturn cloud version the recipe is, to either `saturn_cloud_version` or `recipe_version` because otherwise it's confusing with the image version.

**Comment by jsignell** [comment]
Yeah I like the idea of renaming version to `saturn_version`. I think that would be clearer for recipes as well. Another option might be `schema_version` but I'm fine with either.

**Comment by jnolis** [comment]
Oh tagging @forana since you've been working with recipes too.

**Comment by forana** [comment]
> Yeah I like the idea of renaming version to `saturn_version`. I think that would be clearer for recipes as well. Another option might be `schema_version` but I'm fine with either.

I like `schema_version` - that gives us leeway to decouple it from our releases. AWS does that with their schemas.

**Review by forana — APPROVED** [review]

**Comment by jnolis** [comment]
Okay I updated it to `schema_version` because I agree with that logic. I'm going to merge this and then make a PR to adjust the other types of recipes

**Pull request merged**